pkg/build/options.go

```diff
@@ -158,6 +161,7 @@ func WithCycloneDX() Option {
 func withSBOMber(sbom sbomber) Option {
 	return func(gbo *gobuildOpener) error {
 		gbo.sbom = sbom
+		gbo.sbomType = "sbomber"
```

Comment:
`withSBOMber` is used to pass a custom SBOM impl internally, it shouldn't set the sbomType like this.

This is actually kind of a signal that sbomType isn't a good way to express this. Maybe we can make writing the file part of the sbomer interface's responsibility?